src/macaron/dependency_analyzer/dependency_resolver.py

```diff
@@ -31,9 +29,13 @@ class DependencyTools(str, Enum):
 class DependencyInfo(TypedDict):
     """The information of a resolved dependency."""
 
-    version: str
-    group: str
+    scheme: str
```

Comment:
Instead of duplicating the PURL spec here, wouldn't it be better to store the PURL string directly, which will be parsed using the `packageurl` module?

Reply:
If we store the PURL string as well as the old information (in their fields), we have duplicates. If we don't keep the old information, we have to create the PURL from the string every time we want to extract any information from it. Perhaps storing the actual PURL would make the most sense?

Reply:
I agree with the option of storing the actual PURL object.

Reply:
Storing the `packageurl` object sounds good to me.